uaclient/clouds/gcp.py

```diff
@@ -42,11 +42,37 @@ def __init__(self):
     # mypy does not handle @property around inner decorators
     # https://github.com/python/mypy/issues/1362
     @property  # type: ignore
-    @util.retry(HTTPError, retry_sleeps=[1, 2, 5])
+    @util.retry(exceptions.GCPProAccountError, retry_sleeps=[1, 2, 5])
     def identity_doc(self) -> Dict[str, Any]:
-        url_response, _headers = util.readurl(
-            TOKEN_URL, headers={"Metadata-Flavor": "Google"}
-        )
+        try:
+            headers = {"Metadata-Flavor": "Google"}
+            url_response, _headers = util.readurl(TOKEN_URL, headers=headers)
+        except HTTPError as e:
+            body = getattr(e, "body", None)
+            if body is None and hasattr(e, "read"):
+                reader = getattr(e, "read")()
+                body = reader.decode("utf-8")
+            error_desc = None
+            if body:
+                try:
+                    error_details = json.loads(
+                        body, cls=util.DatetimeAwareJSONDecoder
+                    )
+                except ValueError:
+                    error_details = None
+                if error_details:
+                    error_desc = error_details.get("error_description", None)
+            msg = error_desc if error_desc else getattr(e, "reason", None)
+            msg_code = None
+            if error_desc and "service account" in error_desc.lower():
+                msg += "\n" + messages.GCP_SERVICE_ACCT_URL_MESSAGE
+                msg = messages.GCP_SERVICE_ACCT_NOT_ENABLED_ERROR.msg.format(
+                    error_msg=msg
+                )
```

Comment:
the multiple uses of the `msg` variable are confusing here.
If we combine the message variables like I suggest in my other comment, this can be simplified to
```suggestion
                msg = messages.GCP_SERVICE_ACCT_NOT_ENABLED_ERROR.msg.format(
                    error_msg=error_desc
                )
```